File: api/src/main/java/me/devsnox/planetsystem/api/planet/Planet.java
package me.devsnox.planetsystem.api.planet;

import me.devsnox.planetsystem.api.location.PlanetLocation;

import java.util.List;
import java.util.UUID;
import java.util.function.Consumer;

public interface Planet {

    UUID getUniqueID();

    UUID getOwnerUniqueID();


    String getName();

    List<UUID> getMembers();

    byte getSize();

    void setSize(byte size);

    PlanetLocation getSpawnLocation();

    void setSpawnLocation(PlanetLocation planetLocation);

    void load(Consumer<LoadedPlanet> result);

}

File: api/src/main/kotlin/me/devsnox/planetsystem/api/planet/Planet.kt
package me.devsnox.planetsystem.api.planet

import me.devsnox.planetsystem.api.location.PlanetLocation
import java.util.*

interface Planet {

	val uniqueID: UUID

	val ownerUniqueID: UUID

	val name: String

	val members: MutableList<UUID>

	/**
	 * Change size and if loaded inner region
	 */
	var size: Byte

	var spawnLocation: PlanetLocation

	fun load(result: (LoadedPlanet) -> Unit)

}
